enforcer.go

```diff
@@ -460,44 +466,33 @@ func (e *Enforcer) enforce(matcher string, explains *[]string, rvals ...interfac
 				return false, err
 			}
 
-			switch result := result.(type) {
-			case bool:
-				if !result {
-					policyEffects[i] = effect.Indeterminate
-					continue
-				}
-			case float64:
-				if result == 0 {
-					policyEffects[i] = effect.Indeterminate
-					continue
-				} else {
-					matcherResults[i] = result
-				}
-			default:
-				return false, errors.New("matcher result should be bool, int or float")
+			var eft effect.Effect
+			if !result.(bool) {
+				eft = effect.Indeterminate
+			} else {
+				eft = effect.Allow
 			}
 
-			if j, ok := parameters.pTokens["p_eft"]; ok {
-				eft := parameters.pVals[j]
-				if eft == "allow" {
-					policyEffects[i] = effect.Allow
-				} else if eft == "deny" {
-					policyEffects[i] = effect.Deny
+			if eft == effect.Indeterminate {
+				eft = effect.Indeterminate
+			} else if j, ok := parameters.pTokens["p_eft"]; ok {
+				pEft := parameters.pVals[j]
+				if pEft == "allow" {
+					eft = effect.Allow
+				} else if pEft == "deny" {
+					eft = effect.Deny
 				} else {
-					policyEffects[i] = effect.Indeterminate
+					eft = effect.Indeterminate
 				}
 			} else {
-				policyEffects[i] = effect.Allow
+				eft = effect.Allow
 			}
 
-			if e.model["e"]["e"].Value == "priority(p_eft) || deny" {
+			if eftStream.PushEffect(eft) {
```

Comment:
A good suggestion is to record the current policy  to `explains ` when PushEffect returns `res == true && done == true`.

Reply:
I think it won't affect the performance anyways.